backend/session/session.c

```diff
@@ -220,8 +220,8 @@ static int open_if_kms(struct wlr_session *restrict session, const char *restric
 		goto out_fd;
 	}
 
-	if (res->count_crtcs <= 0 || res->count_connectors <= 0 ||
-		res->count_encoders <= 0) {
+	if (res->count_crtcs < 0 || res->count_connectors < 0 ||
+		res->count_encoders < 0) {
```

Comment:
I don't think those values would ever be negative.
You may as well just remove the check entirely.